src/ui_interface.h

```diff
@@ -102,7 +105,7 @@ class CClientUIInterface
     ADD_SIGNALS_DECL_WRAPPER(NotifyAlertChanged, void, );
 
     /** A wallet has been loaded. */
-    ADD_SIGNALS_DECL_WRAPPER(LoadWallet, void, std::shared_ptr<CWallet> wallet);
+    ADD_SIGNALS_DECL_WRAPPER(LoadWallet, void, std::unique_ptr<interfaces::Wallet>& wallet);
```

Comment:
To make this bouncing around of unique pointers easier to follow (another commit could) move `std::string name = wallet->getWalletName();` all the way up in `WalletModel* WalletController::getOrCreateWallet`, and then add a comment that this is the last time it needs the unique pointer `std::unique_ptr<interfaces::Wallet> wallet`.

Alternatively, but that's too much behavior change for this PR, we could use the wallet name (or some canonical identifier) for the signal.

cc @promag

Reply:
I think this is fine.